ably/realtime_conn.go

```diff
@@ -151,13 +164,45 @@ func (c *Connection) connectWithRecovery(result bool, connKey string, connSerial
 		query.Set(k, v)
 	}
 	if err := c.auth.authQuery(query); err != nil {
+		return nil, err
+	}
+	switch mode {
+	case ResumeMode:
+		query.Set("resume", c.key)
+		query.Set("connectionSerial", fmt.Sprint(c.serial))
+	case RecoveryMode:
+		m := strings.Split(c.recoveryKey, ":")
+		if len(m) > 0 {
```

Comment:
We should error out if the recovery key is malformed. It _needs_ to have three parts, separated by `:`.

Also, we're missing this ([RTN16f](https://docs.ably.io/client-lib-development-guide/features/#RTN16f)):

> The msgSerial component of the recoveryKey [...] is used to set the library internal msgSerial. (If the recover fails, the counter should be reset to 0 per RTN15c3 )

Reply:
addressed in 4449023b755bb04b26b63f4dab7c83ccd5dba667